aldryn_events/cms_plugins.py

```diff
@@ -35,25 +67,24 @@ class UpcomingPlugin(CMSPluginBase):
     form = UpcomingPluginForm
 
     def render(self, context, instance, placeholder):
-        # translated filter the events, language set current language
-        language = get_language_from_request(context['request'],
-                                             check_path=True)
-        namespace = instance.app_config_id and instance.app_config.namespace
         self.render_template = (
             'aldryn_events/plugins/upcoming/%s/upcoming.html' % instance.style
         )
-        context['instance'] = instance
-        # check if we can reverse list view for configured namespace
-        # if no prepare a message to admin users.
-        if not is_valid_namespace(namespace):
-            # add message, should be properly handled in template
-            context['plugin_configuration_error'] = NO_APPHOOK_ERROR_MESSAGE
+        context = super(UpcomingPlugin, self).render(context, instance,
+                                                     placeholder)
+        if context.get('plugin_configuration_error') is not None:
             return context
 
+        context['instance'] = instance
+        language = self.get_language(context['request'])
+        namespace = self.get_namespace(instance)
+        site_id = getattr(get_current_site(context['request']), 'id')
```

Comment:
Might be worth to fallback to `None` if `get_current_site()` returns something different from the `Site` model instance.